### PR TITLE
fix(expect,internal,testing): support `expect.assertions`

### PR DESCRIPTION
**Pull request by eryue0220**
Part of #3964, support `expect.assertions` api

### CODE REVIEW

File: internal/assertion_state.ts [hunk 1]
```diff
@@ -69,26 +142,56 @@ export class AssertionState {
...
+   *
+   * const assertionState = new AssertionState();
+   * if (assertionState.checkAssertionCountSatisfied()) {
+   *   // throw AssertionError("");
+   * }
+   * ```
+   */
+  checkAssertionCountSatisfied(): boolean {
+    return this.#state.assertionCount !== undefined &&
+      this.#state.assertionCount > this.#state.assertionTriggeredCount;
```
Comment (kt3k) [inline internal/assertion_state.ts:193]: It seems in jest, `expect.assertions(N)` call expects exactly `N` assertions are made. For example, the below test file fails with the error message `Expected one assertion to be called but received two assertion calls.`:

```js
test("test", () => {
  expect.assertions(1);
  expect(1).toBe(1);
  expect(2).toBe(2);
});
```

Comment (eryue0220) [reply internal/assertion_state.ts:193]: I'll check this later.

Comment (eryue0220) [reply internal/assertion_state.ts]: Thanks, I've updated.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**